ste/xfer-anyToRemote.go

```diff
@@ -30,12 +31,83 @@ import (
 	"sync"
 
 	"github.com/Azure/azure-pipeline-go/pipeline"
+	"github.com/Azure/azure-storage-blob-go/azblob"
+
 	"github.com/Azure/azure-storage-azcopy/common"
 )
 
 // This sync.Once is present to ensure we output information about a S2S access tier preservation failure to stdout once
 var s2sAccessTierFailureLogStdout sync.Once
 
+// This sync.Once and string pair ensures that we only get a user's destination account kind once when handling set-tier
+// Premium block blob doesn't support tiering, and page blobs only support P1-80.
+// There are also size restrictions on tiering.
+var destAccountSKU string
+var destAccountKind string
+var tierSetPossibleFail bool
+var getDestAccountInfo sync.Once
+var getDestAccountInfoError error
+
+func prepareDestAccountInfo(bURL azblob.BlobURL, jptm IJobPartTransferMgr, ctx context.Context, mustGet bool) {
+	getDestAccountInfo.Do(func() {
+		infoResp, err := bURL.GetAccountInfo(ctx)
+		if err != nil {
+			// If GetAccountInfo fails, this transfer should fail because we lack at least one available permission
+			// UNLESS the user is using OAuth. In which case, the account owner can still get the info.
+			// If we fail to get the info under OAuth, don't fail the transfer.
+			// (https://docs.microsoft.com/en-us/rest/api/storageservices/get-account-information#authorization)
+			if mustGet {
+				getDestAccountInfoError = err
+			} else {
+				tierSetPossibleFail = true
+				destAccountSKU = "failget"
+				destAccountKind = "failget"
+			}
+		} else {
+			destAccountSKU = string(infoResp.SkuName())
+			destAccountKind = string(infoResp.AccountKind())
+		}
+	})
+
+	if getDestAccountInfoError != nil {
+		jptm.FailActiveSendWithStatus("Checking destination tier availability (Set PageBlob tier) ", getDestAccountInfoError, common.ETransferStatus.TierAvailabilityCheckFailure())
+	}
+}
+
+// TODO: Infer availability based upon blob size as well, for premium page blobs.
+func BlobTierAllowed(destTier azblob.AccessTierType) bool {
```

Comment:
>BlobTierAllowed [](start = 5, length = 15)

Normally I don't recommend this kind of client-side logic, but in this case I think it's fine since we can only have new rules with new service versions, i.e. no cx will be blocked because the service rolled out a new rule